lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -175,12 +188,12 @@ export const migrateGranuleAndFilesViaTransaction = async (
     if (error instanceof RecordAlreadyMigrated) {
       granulesResult.skipped += 1;
     } else {
+      const errorMessage = `Could not create granule record and file records in RDS for DynamoDB Granule granuleId: ${dynamoRecord.granuleId} with files ${JSON.stringify(dynamoRecord.files)}`;
       granulesResult.failed += 1;
       filesResult.failed += files.length;
-      logger.error(
-        `Could not create granule record and file records in RDS for DynamoDB Granule granuleId: ${dynamoRecord.granuleId} with files ${dynamoRecord.files}`,
-        error
-      );
+
+      stream.write(JSON.stringify(`${errorMessage}, Cause: ${error}`));
```

Comment:
I think the JSON.stringify here is unnecessary. Part of what the JSONStream is doing for you is this stringification automatically.

With the JSON.stringify in place, when I inspect the local contents of the error file, I see:

```
{
  "errors": [
    "\"Could not create granule record and file records in RDS for DynamoDB Granule granuleId: 9a2c4 with files [{\\\"bucket\\\":\\\"ba67b2bd6a\\\",\\\"fileName\\\":\\\"MOD09GQ.A4369670.7bAGCH.006.0739896140643.hdf\\\",\\\"size\\\":1098034,\\\"checksum\\\":\\\"checkSum01\\\",\\\"checksumType\\\":\\\"md5\\\",\\\"source\\\":\\\"source\\\",\\\"type\\\":\\\"data\\\",\\\"key\\\":\\\"fc75692596\\\"}], Cause: RecordDoesNotExist: Record in collections with identifiers {\\\"name\\\":\\\"6fe57\\\",\\\"version\\\":\\\"0.0.0\\\"} does not exist.\"",
    "\"Could not create granule record and file records in RDS for DynamoDB Granule granuleId: f0095 with files [{\\\"bucket\\\":\\\"ba67b2bd6a\\\",\\\"fileName\\\":\\\"MOD09GQ.A4369670.7bAGCH.006.0739896140643.hdf\\\",\\\"size\\\":1098034,\\\"checksum\\\":\\\"checkSum01\\\",\\\"checksumType\\\":\\\"md5\\\",\\\"source\\\":\\\"source\\\",\\\"type\\\":\\\"data\\\",\\\"key\\\":\\\"851a0de80f\\\"}], Cause: RecordDoesNotExist: Record in collections with identifiers {\\\"name\\\":\\\"dd859\\\",\\\"version\\\":\\\"0.0.0\\\"} does not exist.\""
  ]
}
```

You can see that it is unnecessarily double-escaping the contents of each error message, which is less readable.

With the JSON.stringify removed, I see:

```
{
  "errors": [
    "Could not create granule record and file records in RDS for DynamoDB Granule granuleId: a7c71 with files [{\"bucket\":\"e824412d07\",\"fileName\":\"MOD09GQ.A4369670.7bAGCH.006.0739896140643.hdf\",\"size\":1098034,\"checksum\":\"checkSum01\",\"checksumType\":\"md5\",\"source\":\"source\",\"type\":\"data\",\"key\":\"f7d6ebe20e\"}], Cause: RecordDoesNotExist: Record in collections with identifiers {\"name\":\"a07de\",\"version\":\"0.0.0\"} does not exist.",
    "Could not create granule record and file records in RDS for DynamoDB Granule granuleId: 24935 with files [{\"bucket\":\"e824412d07\",\"fileName\":\"MOD09GQ.A4369670.7bAGCH.006.0739896140643.hdf\",\"size\":1098034,\"checksum\":\"checkSum01\",\"checksumType\":\"md5\",\"source\":\"source\",\"type\":\"data\",\"key\":\"90520d5818\"}], Cause: RecordDoesNotExist: Record in collections with identifiers {\"name\":\"6207d\",\"version\":\"0.0.0\"} does not exist."
  ]
}
```

Which is I think more what we want